.github/workflows/main.yml

```
@@ -23,14 +23,33 @@ jobs:
     name: Build, lint, and test
     uses: ./.github/workflows/build-lint-test.yml
 
-  all-jobs-pass:
-    name: All jobs pass
+  all-jobs-completed:
+    name: All jobs completed
     runs-on: ubuntu-latest
     needs:
       - check-workflows
       - build-lint-test
+    outputs:
+      PASSED: ${{ steps.set-output.outputs.PASSED }}
+    steps:
+      - name: Set PASSED output
+        id: set-output
+        run: echo "PASSED=true" >> "$GITHUB_OUTPUT"
+
+  all-jobs-pass:
+    name: All jobs pass
+    if: ${{ always() }}
+    runs-on: ubuntu-latest
+    needs: all-jobs-completed
     steps:
-      - run: echo "Great success!"
+      - name: Check that all jobs have passed
+        run: |
+          passed="${{ needs.all-jobs-completed.outputs.PASSED }}"
+          if [[ $passed == "true" ]]; then
+            exit 0
+          else
+            exit 1
+          fi
```

not a blocker but perhaps redundant with explicit `exit 0`?
```suggestion
          if [[ $passed != "true" ]]; then
            exit 1
          fi
```
or even
```suggestion
          [[ $passed != "true" ]] && exit 1
```

---

Good point!